src/parse/asp/util.go

```diff
@@ -55,3 +58,78 @@ func FindArgument(statement *Statement, args ...string) *CallArgument {
 	}
 	return nil
 }
+
+// StatementLite is a simplified version of asp.Statement
+// Here we only care about statement with Idents and Expressions
+// In the future we can also add more fields in this struct
+type StatementLite struct {
+	Statement  *Statement
+	Expression *Expression
+}
+
+// StatementFromAst recursively finds asp.IdentStatement and asp.Expression in the ast
+// and returns a valid statement pointer if within range
+func StatementFromAst(v reflect.Value, position Position) (*StatementLite, error) {
```

Comment:
Hmmm on second thought - rather than having the custom type, just return `(*Statement, *Expression, error)`

In fact, do you need the error? I can't see a branch where it is non-nil?